### PR TITLE
implementation of AA models

### PR DESCRIPTION
**Pull request by GertjanBisschop**
solves #1023

### CODE REVIEW

**Comment by GertjanBisschop** [comment]
These are the same checks for the AA models as for the nucleotide models:

[AA.pdf](https://github.com/tskit-dev/msprime/files/4752186/AA.pdf)

`validate_model(model)` and `validate_stationary(model)` still needs to be added to `tests/test_mutations.py`. But I got an error that I didn't have before: `ValueError: 'reference_size' not specified`. Something similar seems to have been addressed a couple of days back... I will check the slack channel.

File: msprime/mutations.py [hunk 1]
```diff
@@ -187,6 +209,466 @@ def __init__(
         super().__init__(alleles, root_distribution, transition_matrix)
 
 
+class BLOSUM62(MutationModel):
+    def __init__(self):
+        alleles = _AMINO_ACIDS
+        num_alleles = len(alleles)
```
Comment (jeromekelleher) [inline msprime/mutations.py:215]: I would be good to document where these values came from - can you give some notes here on where you got them? Some of the values are at 3 digits of precision, others at 12. Good to give some provenance on that also.

File: msprime/mutations.py [hunk 2]
```diff
@@ -187,6 +209,466 @@ def __init__(
...
+        row_sums = transition_matrix.sum(axis=1, dtype="float64")
+        np.fill_diagonal(transition_matrix, 1.0 - row_sums)
+        super().__init__(alleles, root_distribution, transition_matrix)
+
+
+class PAM(MutationModel):
+    def __init__(self):
+        alleles = _AMINO_ACIDS
+        num_alleles = len(alleles)
+        root_distribution = [
```
Comment (jeromekelleher) [inline msprime/mutations.py:446]: Likewise here on the numbers - where do they come from?

File: verification.py [hunk 3]
```diff
@@ -2107,10 +2107,12 @@ def run_xi_beta_breakpoints(self):
                 f"n=100_alpha={alpha}",
                 sample_size=100,
                 Ne=Ne,
-                r=1e-7,
```
Comment (jeromekelleher) [inline verification.py:2110]: Looks like something funny has happened here while you were rebasing. This is cancelling out some of the changes made in #1059. Probably easiest to inspect the code and manually revert your local changes.

File: msprime/mutations.py (hunk 1 above)
Comment (GertjanBisschop) [reply msprime/mutations.py:215]: True. These are the values that are used in both pyvolve and seqgen for their rate matrices. I still need to check whether there is a better way, meaning deriving the ratet matrix from the time-dependent probability matrices as originally published. And whether it makes sense to actually do this each time, or if the best way is to store the raw values as is done in pyvolve and seq-gen.

Comment (jeromekelleher) [reply msprime/mutations.py:215]: Copying in the values like this is totally fine - just a note on where they came from is all we need.

**Comment by GertjanBisschop** [comment]
PAM:
root distribution
root_distribution

diversity (with scikit)
pi

transition probabilities
alleles

BLOSUM62:
root distribution
root_distribution

diversity
pi

transition probabilities
alleles

**Comment by jeromekelleher** [comment]
Great, thanks @GertjanBisschop! There's still two accidental changes to the multiple merger coalescent verification stuff - we're good to merge once these are reverted.

**Comment by GertjanBisschop** [comment]
No, I think I have now changed the corrections back to the mistake. I can't see what else is still wrong.

**Comment by jeromekelleher** [comment]
Great, thanks @GertjanBisschop! Can you [squash](https://stdpopsim.readthedocs.io/en/latest/development.html#rebasing) the commits down to 1 please? Ready to merge then.

**Comment by GertjanBisschop** [comment]
@jeromekelleher, forgot to ping you. All commits squashed down to 1.

**Comment by jeromekelleher** [comment]
Merging; thanks @GertjanBisschop!

**Comment by jeromekelleher** [comment]
Dammit, looks like this is failing because of #1060. I'll fix this.

**Comment by jeromekelleher** [comment]
Thanks again @GertjanBisschop! Just FYI, if you want GitHub to autoclose issues you need to use "Closes" or "Fixes". "solves" doesn't work, it seems.

https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by GertjanBisschop** [comment]
OK thanks. Good to know. Shall I add the 'closed in' comment to the other mutation model issues as well?

**Comment by jeromekelleher** [comment]
Yes please!

**Pull request merged**